### PR TITLE
Check special char

### PR DESCRIPTION
**Pull request by yann300**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -369,7 +369,12 @@ function run () {
   // Add files received from remote instance (i.e. another browser-solidity)
   function loadFiles (filesSet) {
     for (var f in filesSet) {
-      filesProviders['browser'].set(helper.createNonClashingName(f, filesProviders['browser']), filesSet[f].content)
+      var name = helper.createNonClashingName(f, filesProviders['browser'])
+      if (filesProviders['browser'].checkSpecialChars(name)) {
```
Comment (chriseth) [inline src/app.js]: While this is a good counter-measure, I don't think this is a proper fix. The code using yoyo has to do proper escaping.

File: src/app/files/shared-folder.js [hunk 2]
```diff
@@ -45,6 +45,10 @@ class SharedFolder {
     cb()
   }
 
+  checkSpecialChars (name) {
+    return name.match(/(\/|:|\*|\?|"|<|>|\\|\||')/) != null
```
Comment (chriseth) [inline src/app/files/shared-folder.js]: Can you create this function only once?

Comment (chriseth) [reply src/app/files/shared-folder.js]: Ah and something like `/[/:*?"<>\\\|]/` would be better I think.

File: src/app/tabs/settings-tab.js [hunk 3]
```diff
@@ -140,6 +140,10 @@ function loadVersion (version, queryParams, appAPI, el) {
 
     url = location + 'soljson.js'
   } else {
+    if (version.indexOf('soljson') !== 0 || version.match(/(\/|:|\*|\?|"|<|>|\\|\||')/) != null) {
```
Comment (chriseth) [inline src/app/tabs/settings-tab.js]: A third version of the regex :-)

File: src/app/files/shared-folder.js (hunk 2 above)
Comment (yann300) [reply src/app/files/shared-folder.js]: one is for browser-storage and the other for shared-folder (remixd). there might have a different requirement for `checkSpecialChars` ... but I guess that is fine to keep the same behavior for both

File: src/app/tabs/settings-tab.js [hunk 4]
```diff
@@ -140,6 +141,10 @@ function loadVersion (version, queryParams, appAPI, el) {
 
     url = location + 'soljson.js'
   } else {
+    if (version.indexOf('soljson') !== 0 || helper.checkSpecialChars(version) != null) {
```
Comment (chriseth) [inline src/app/tabs/settings-tab.js]: why do you compare to `null`?

Comment (yann300) [reply src/app/tabs/settings-tab.js]: yes I guess `if (version.indexOf('soljson') !== 0 || helper.checkSpecialChars(version))` would have been sufficient...

**Pull request merged**